GoodreadsEDA/scp_bks.py

```diff
@@ -104,97 +78,124 @@ def print_pg_number(page_count):
-    except NoSuchElementException as e:
-        print("No Score Given")
-        score = "None"
+        score = scores_to_numbers_dict[score]
+    except NoSuchElementException:
+        # print("No Score Given")
+        score = None
     # Convert Score to Number Value
-    score = scores_to_numbers(score)
-    print(f"| {user} | {date} | {score}")
-    return (user, date, score)
-
-def insert_into_review_table(c, author, title, url,
-                             score, user, date):
-    """Inserts all of the avilable data pulled from the review in to the "Review"
-    table in the databse."""
+    # print(f"| {user} | {date} | {score}")
+    return user, date, score
 
-    try:
-        c.execute(f"""INSERT INTO Reviews (
-                     book_auth, book_title,
-                     book_url, review_score,
-                     user_name, review_date)
-                     VALUES ("{author}", "{title}",
-                     "{url}", "{score}",
-                     "{user}", "{date}"
-                     )""")
-    except sqlite3.OperationalError as e:
-        print(e)
-        c.execute(f"""INSERT INTO Reviews (
-                 book_auth, book_title,
-                 book_url, review_score,
-                 user_name, review_date)
-                 VALUES ("{author}", "{title}",
-                 "{url}", "{score}",
-                 "invalid_username", "{date}"
-                 )""")
 
+def insert_into_review_table(c, author, title, url, score, user, date):
+    """Inserts all of the available data pulled from the review in to the "Reviews"
+    table in the database."""
 
-def main():
-    # Get a list of all the urls for the books by the authors supplied from
-    # mk_bklst.py, and remove the header row.
-    book_urls = get_book_urls()
-    book_urls.pop(0)
-
-    # Options for the Chrome driver, so Chrome operates without calling a window.
-    options = webdriver.ChromeOptions()
-    options.add_argument('headless')
-
-    # Connect to db, and create new table "Reviews"
-    conn = sqlite3.connect(f"{os.getcwd()}/review_dbs/reviews.db")
-    c = conn.cursor()
-    create_review_table(c)
-
-    for entry in book_urls:
+    # escapes all non-alphanumeric characters for consistent insertion
+    user = re.escape(user)
+    try:
+        c.execute("""INSERT INTO Reviews (book_auth, book_title, book_url, review_score, user_name, review_date)
+                     VALUES (?, ?, ?, ?, ?, ?)""", (author, title, url, score, user, date))
+    except sqlite3.OperationalError:
+        # this should actually never happen now, but I'm leaving it here for good measure
+        traceback.print_exc()
+        c.execute("""INSERT INTO Reviews (book_auth, book_title, book_url, review_score, user_name, review_date)
+                     VALUES (?, ?, ?, ?, 'invalid username', ?)""", (author, title, url, score, date))
+
+
+def print_scraping_status(current_book, total_books, start_timer):
```

Comment:
This was just to give a sense of how long the process was going to take, as it wasn't super clear from the previous prints. And also because it takes a very long time.